.circleci/scripts/cache.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set -euo pipefail
+
+# Cache is available only on our custom runner
+if [ "$CIRCLECI_RUNNER_CLASS" != "spaceshard/ax41" ]; then
+    echo "Custom cache available only on self-hosted spaceshard/ax41 runner"
+    exit 0
+fi
+
+if [ $# -ne 1 ]; then
+    echo "Please provide cache action as an argument. (save, load or cleanup)"
+    exit 1
+fi
+
+action="$1"
+
+# VARIABLES
+# Files to use as cache key
+cache_key_files=("Cargo.lock" "rust-toolchain.toml") 
+
+ # Directories to cache
+cached_dirs=("target/release/.fingerprint target/release/build target/release/deps")
```

Comment:
We used to cache paths that are outside of the repository. Also, we cached compilation artifacts from both `release` and `debug` mode:
https://github.com/0xSpaceShard/starknet-devnet-rs/commit/01fb56e24c8b86f6cb8e306cd6221e7a5c415f18#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L80-L88

Why is this no longer necessary?

Reply:
- On this runner, global cargo and rustup files are preserved in between runs (toolchains for example)
 - If you look closely, no step before "save cache" does produce debug artifacts so there is no need to save them

Reply:
> * On this runner, global cargo and rustup files are preserved in between runs (toolchains for example)

Makes sense

> * If you look closely, no step before "save cache" does produce debug artifacts so there is no need to save them

I am aware of that, my question is also: "Should we also cache debug artifacts or not"?

Reply:
Ah, sorry. No clue, that is something that I considered out of scope for this PR and I feel that there is somebody else more qualified than me to make informed decision. I just can note that debug files are pretty big so I am not sure of a time advantage that could be gained from caching as compression and decompression of those files might last nontrivial amount of time.